geode-core/src/main/java/org/apache/geode/deployment/JarDeploymentService.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.deployment;
+
+import java.io.File;
+import java.util.List;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.management.api.ClusterManagementService;
+import org.apache.geode.management.configuration.Deployment;
+import org.apache.geode.services.result.ServiceResult;
+
+/**
+ * Implementations of {@link JarDeploymentService} will be responsible for deploying jars into Geode
+ * from both gfsh and the {@link ClusterManagementService}.
+ *
+ * @since Geode 1.15
+ */
+@Experimental
+public interface JarDeploymentService {
```

Comment:
Why is this interface not in an internal package? It seems like it could be in the same package as JarDeploymentServiceFactory which is internal. Does some other non-internal api expose this interface? Would a geode user want to implement this interface? By making it internal you don't need to worry about marking it experimental because internal apis are free to change or be removed.

Reply:
It can be internal, we will move it.

Reply:
Fixed.